server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -768,6 +811,41 @@ private void close() {
         }
     }
 
+    private static class EncodeLengthTrackingChunkedRestResponseBody implements ChunkedRestResponseBody {
+
+        private final ChunkedRestResponseBody delegate;
+        private final Consumer<Integer> encodedLengthConsumer;
+        private int encodedLength = 0;
```

Comment:
This is a good point. It is now handled with `Math.addExact`. The value is also capped to Integer.MAX_VALUE in case of overflow. I think it is OK to not track the exact number since (I think?) it is rare and we don't really need the exact number for histogram. The total response size will be off. But I feel that's OK.
Please see code changes [here](https://github.com/elastic/elasticsearch/pull/99852/commits/5849e3d1e3752df71fe6e89fab0584b4f3acbcf2#diff-214f5374e07a22c0e5d0d845cd7a890a4c31b22bef100fa609b4c74289be1681R840-R848))

Reply:
Hm I think that's going to be trappy, we might well want to add another bucket to the end of the histogram later. Why not just use `long` here?

Reply:
I didn't think it is a case that requires accurate handling. But upgrading to long is also fine. I updated it in 49b95f8765af2801adb76fee643fb104baa8a5da